src/dependencies.jl

```diff
@@ -822,8 +822,16 @@ macro install (_libmaps...)
                         """
                         # Macro to load a library
                         macro checked_lib(libname, path)
-                            (dlopen_e(path) == C_NULL) && error("Unable to load \\n\\n\$libname (\$path)\\n\\nPlease re-run Pkg.build(package), and restart Julia.")
-                            quote const \$(esc(libname)) = \$path end
+                            if dlopen_e(path) == C_NULL
+                                depsfile = joinpath(Pkg.dir(pkgname), "deps/deps.jl")
+                                if myid() != 1 && isfile(depsfile)
+                                    path = eval(include_string(readall(depsfile)))
+                                else
+                                    error("Unable to load \\n\\n\$libname (\$path) on $(gethostname())\\n\\nPlease re-run Pkg.build(package), and restart Julia.")
```

Comment:
I don't think you should print the hostname here. That's not the right layer to handle this kind of information: the host on which the error occurred should be reported by the parallel support code.